src/adaptor.py

```diff
@@ -331,6 +331,15 @@ def read_from_fs(path, **kwargs):
     outgoing = fs_adaptor.OutgoingQueue()
     return incoming, outgoing
 
+def read_from_s3(path, kwargs):
```

Comment:
I'm not entirely sure, but would `kwargs` possibly conventionally be `**kwargs` in this situation?

Reply:
Guessing maybe since you pass it later, you do not want to "unpack" the values early?

Reply:
I see what you're saying, it could be either-or. I prefer to avoid the `*` and `**` operators these days.